doc/source/architecture.rst

```diff
@@ -0,0 +1,15 @@
+Project Architecture
+********************
+
+This is the class inheritance diagram for the key pieces of ophyd. This is
+a selection meant to give a readable and representative picture of the
+package's organization.
+
+.. inheritance-diagram:: ophyd.Device ophyd.Component ophyd.EpicsSignal ophyd.EpicsSignalRO ophyd.Signal ophyd.EpicsMotor ophyd.EpicsScaler ophyd.EpicsMCA ophyd.AreaDetector ophyd.HDF5Plugin ophyd.DynamicDeviceComponent ophyd.PVPositioner ophyd.SingleTrigger
```

Comment:
You only need to list the leaves, it will build the rest of the graph.  Looks like there are knobs to change both size and direction of diagram, but no obvious way to touch them from rst...